### PR TITLE
Adds HorizontalContentCardConnected and Component prop to ContentCardConnected

### PR DESCRIPTION
**Pull request by burkeshartsis**
## DESCRIPTION

This adds a `Component` prop to `ContentCardConnected` and changes `ContentCardConnected`s default rendered child to `DefaultCard`. Additionally, it adds a `HorizontalContentCardConnected` component. Lastly, we add some missing exports for some of the newer ui-kit cards 🙃 

### What does this PR do, or why is it needed?
This makes `ContentCardConnected` way more flexible by letting us pass in other cards to the same query. All of our news cards share the same same basic API so this is really easy 💥

Our new card designs also need to be rendered at the right time. This adds a mapper HOC that selects the right card for the right place.

### What design trade-offs/decisions were made?
TONS.

1. This PR breaks things. In particular horizontal card rendering is pretty broken. 
2. We also no longer render 1 card in two ways (vertical vs horizontal scroll). Instead we have separate components for that and there is fair amount of legacy `tile` props still floating around that probably could be refactored.
3. Merging this means committing otherwise broken snapshots to 😢
4. The new cards require images and in a lot of cases titles so there are a bunch of prop warnings.
5. no tests and no stories 😭

### How do I test this PR?
Check that the app now renders the new card styles in the feeds. This PR could use stories for `ContentCardConnected` and `HorizontalContentCardConnected` but they didn't exist before this so...

### What automated tests did you write?
This PR could use tests for `ContentCardConnected` and `HorizontalContentCardConnected` but they didn't exist before this so...

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Partial refactor work associated with #852 
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apolloschurchapp/src/ui/ContentCardConnected/contentCardComponentMapper.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { DefaultCard, HighlightCard } from '@apollosproject/ui-kit';
+
+const contentCardComponentMapper = (props) => {
+  // check if we have a custom Component prop to use or we'll use the default Component prop.
+  let ComponentToRender;
+
+  // map typename to the the card we want to render.
+  switch (get(props, '__typename')) {
+    case 'MediaContentItem':
+      ComponentToRender = HighlightCard;
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/ui/ContentCardConnected/contentCardComponentMapper.js]: You can condense this `switch` statement by removing all but the bottom assignment :) I would also recommend `return`ing inside the case, I find it's easier to parse a switch statement that way vs assignment. Also then you don't need to `break`!

File: packages/apolloschurchapp/src/ui/HorizontalContentCardConnected/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    />
+  )
+);
+
+HorizontalContentCardConnected.propTypes = {
+  Component: PropTypes.func,
+  isLoading: PropTypes.bool,
+};
+
+HorizontalContentCardConnected.defaultProps = {
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/ui/HorizontalContentCardConnected/index.js]: 👍

**Comment by burkeshartsis** [comment]
@vinnyjth what if instead of defaulting the `Component` prop on `ContentCardConnected` to `contentCardComponentMapper` we exported `contentCardComponentMapper` and passed it in as the `Component` prop when using `ContentCardConnected` that way `ContentCardConnected can always just render a single component and instead we are just extending that functionality via this other function when needed. Granted I’m not sure when we wouldn’t need it but… thoughts?

**Comment by vinnyjth** [comment]
I'm fine the way we are doing it now. Seems the most "sane"

**Comment by vinnyjth** [comment]
Huge step in the right direction. Awesome simplification. Enhanced modularity. I've approved this PR once and I'll do it again!

**Pull request merged**